In [0]:
import pandas as pd
import pandas
import requests
import json
import urllib 
from urllib.request import Request, urlopen

In [0]:
DATA_URL = 'https://s3.amazonaws.com/vargo.aprd6342/data/msas.csv'
df = pd.read_csv(DATA_URL)
df.head(10)

CBSA                               NAME  POPESTIMATE2017
0  10180                        Abilene, TX           170219
1  10420                          Akron, OH           703505
2  10500                         Albany, GA           151434
3  10540                         Albany, OR           125047
4  10580        Albany-Schenectady-Troy, NY           886188
5  10740                    Albuquerque, NM           910726
6  10780                     Alexandria, LA           153984
7  10900  Allentown-Bethlehem-Easton, PA-NJ           840550
8  11020                        Altoona, PA           123457
9  11100                       Amarillo, TX           264925

In [0]:
# seperate the data as 2 parts, then change the type
geoid = "31000US" + df["CBSA"].astype(str)
half = geoid[:len(geoid)//2]
rest = geoid[len(geoid)//2:]
# join list together in a way that the api accepts
half = ','.join(half)
rest = ','.join(rest)

# Question 1:What city has the highest count of the audience you chose for parameter #1?

In [0]:
# make a python list of all the table ids you want to download
tableidsQ1 = ['B19001']
# join list together in a way that the api accepts
tableidstringQ1 = ','.join(tableidsQ1)

In [0]:
# iterate over each of your metro areas and get the right data for each metro
url = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tableidstringQ1, half)
loadedjson = requests.get(url)
parsedjson1 = json.loads(loadedjson.text)
url = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tableidstringQ1, rest)
loadedjson = requests.get(url)
parsedjson2 = json.loads(loadedjson.text)

In [0]:
highest = pd.DataFrame(columns=['country', 'population'], index=list(range(1,len(geoid))))
for i in list(range(1,len(geoid)//2)):
    highest['country'][i]=[geoid[i]]
    highest['population'][i] = parsedjson1['data'][geoid[i]]['B19001']['estimate']['B19001011'] + parsedjson1['data'][geoid[i]]['B19001']['estimate']['B19001012']

In [0]:
for i in list(range(192,len(geoid))):
    highest['country'][i]=[geoid[i]]
    highest['population'][i] = parsedjson2['data'][geoid[i]]['B19001']['estimate']['B19001011'] + parsedjson2['data'][geoid[i]]['B19001']['estimate']['B19001012']

In [0]:
print(highest.loc[highest['population'] == max(highest['population'])])

            country   population
249  [31000US35620]  1.01121e+06


In [0]:
print(df.loc[df['CBSA'] == 35620])

      CBSA                                   NAME  POPESTIMATE2017
249  35620  New York-Newark-Jersey City, NY-NJ-PA         20320876


We should choose $50k to 74.9k which are 'B19001011' and 'B19001012'.

New York-Newark-Jersey City, NY-NJ-PA has the highest count of average house hold income that ranged between 50k to 74.9k, which is 1.01121e+06.

# Question 2: What city has the highest count of the audience you chose for parameter #2?

In [0]:
tableidsQ2 = ['B01001']
# join list together in a way that the api accepts
tableidstringQ2 = ','.join(tableidsQ2)

In [0]:
# iterate over each of your metro areas and get the right data for each metro
url = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tableidstringQ2, half)
loadedjson = requests.get(url)
parsedjson11 = json.loads(loadedjson.text)
url = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tableidstringQ2, rest)
loadedjson = requests.get(url)
parsedjson22 = json.loads(loadedjson.text)

In [0]:
age_range = ['B01001007','B01001008','B01001009','B01001010','B01001011','B01001012']
newdf = pd.DataFrame(columns=['location', 'population'], index=list(range(1,len(geoid))))

In [0]:
# parsed json file
for i in list(range(1,len(geoid)//2)):
    newdf['location'][i]=[geoid[i]]
    newdf['population'][i] = parsedjson11['data'][geoid[i]]['B01001']['estimate'][age_range[0]] + parsedjson11['data'][geoid[i]]['B01001']['estimate'][age_range[1]] + parsedjson11['data'][geoid[i]]['B01001']['estimate'][age_range[2]] + parsedjson11['data'][geoid[i]]['B01001']['estimate'][age_range[3]] + parsedjson11['data'][geoid[i]]['B01001']['estimate'][age_range[4]] + parsedjson11['data'][geoid[i]]['B01001']['estimate'][age_range[5]] 
    
for i in list(range(192,len(geoid))):
    newdf['location'][i]=[geoid[i]]
    newdf['population'][i] = parsedjson22['data'][geoid[i]]['B01001']['estimate'][age_range[0]] + parsedjson22['data'][geoid[i]]['B01001']['estimate'][age_range[1]] + parsedjson22['data'][geoid[i]]['B01001']['estimate'][age_range[2]] + parsedjson22['data'][geoid[i]]['B01001']['estimate'][age_range[3]] + parsedjson22['data'][geoid[i]]['B01001']['estimate'][age_range[4]] + parsedjson22['data'][geoid[i]]['B01001']['estimate'][age_range[5]]

In [0]:
print(newdf.loc[newdf['population'] == max(newdf['population'])])

           location   population
249  [31000US35620]  2.33054e+06


In [0]:
print(df.loc[df['CBSA'] == 35620])

      CBSA                                   NAME  POPESTIMATE2017
249  35620  New York-Newark-Jersey City, NY-NJ-PA         20320876


We should choose age range 18-34:
'B01001007', 'B01001008', 'B01001009', 'B01001010', 'B01001011', 'B01001012'

New York-Newark-Jersey City, NY-NJ-PA has the highest count of age range 18-34, which is 2.33054e+06.

# Question 3: What city has the highest count of the audience you chose for parameter #3?

In [0]:
tableidsQ3 = ['B03003']
# join list together in a way that the api accepts
tableidstringQ3 = ','.join(tableidsQ3)

In [0]:
# iterate over each of your metro areas and get the right data for each metro
url = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tableidstringQ3, half)
loadedjson = requests.get(url)
parsedjson111 = json.loads(loadedjson.text)
url = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tableidstringQ3, rest)
loadedjson = requests.get(url)
parsedjson222 = json.loads(loadedjson.text)

In [0]:
highest2 = pd.DataFrame(columns=['location', 'population'], index=list(range(1,len(geoid))))

In [0]:
# parsed json file
for i in list(range(1,len(geoid)//2)):
    highest2['location'][i]=[geoid[i]]
    highest2['population'][i] = parsedjson111['data'][geoid[i]]['B03003']['estimate']['B03003003'] 
for i in list(range(192,len(geoid))):
    highest2['location'][i]=[geoid[i]]
    highest2['population'][i] = parsedjson222['data'][geoid[i]]['B03003']['estimate']['B03003003']

In [0]:
print(highest2.loc[highest2['population'] == max(highest2['population'])])

           location   population
210  [31000US31080]  6.00354e+06


In [0]:
print(df.loc[df['CBSA'] == 31080])

      CBSA                                NAME  POPESTIMATE2017
210  31080  Los Angeles-Long Beach-Anaheim, CA         13353907


The Hispanic code would be B03003003.
Los Angeles-Long Beach-Anaheim, CA has the highest count of hispanic, which is 6.00354e+06.

# Question 4:What city has the highest mean percentage across all three categories?

In [0]:
# create new columns and calculate the percentages
HH_Income_Percentage = []
Pop_by_age_Percentage = []
Pop_by_Hispanic_Percentage = []
for i, row in highest.iterrows():
    Percentage_Income = highest['population'][i]/max(highest['population'])
    Percentage_Age = newdf['population'][i]/max(newdf['population'])
    Percentage_Hispanic = highest2['population'][i]/max(highest2['population'])
    HH_Income_Percentage.append(Percentage_Income)
    Pop_by_age_Percentage.append(Percentage_Age)
    Pop_by_Hispanic_Percentage.append(Percentage_Hispanic)

In [0]:
highest['percentage'] = HH_Income_Percentage
newdf['percentage'] = Pop_by_age_Percentage
highest2['percentage'] = Pop_by_Hispanic_Percentage

In [0]:
Highmean = []
for i, row in highest.iterrows():
    Mean = (highest['percentage'][i] + newdf['percentage'][i]+ highest2['percentage'][i])/3
    Highmean.append(Mean)
highest['percentage mean'] = Highmean

In [0]:
print(highest.loc[highest['percentage mean'] == max(highest['percentage mean'])])

            country   population  percentage  percentage mean
249  [31000US35620]  1.01121e+06         1.0         0.940082


In [0]:
print(df.loc[df['CBSA'] == 35620])

      CBSA                                   NAME  POPESTIMATE2017
249  35620  New York-Newark-Jersey City, NY-NJ-PA         20320876


New York-Newark-Jersey City, NY-NJ-PA has the highest percentage across all three categories, which is 0.940082.